Terminal.Gui/Views/ScrollBarView.cs

```diff
@@ -204,54 +120,47 @@ public int Position
         get => _position;
         set
         {
-            _position = value;
-
             if (IsInitialized)
             {
                 // We're not initialized so we can't do anything fancy. Just cache value.
                 SetPosition (value);
             }
+            else
+            {
+                _position = value;
+            }
         }
     }
 
     // BUGBUG: v2 - Why can't we get rid of this and just use Visible?
+    // We need this property to distinguish from Visible which will also affect the parent
```

Comment:
The `VisibleChanged` event is really not necessary and I will remove it in this PR.

Reply:
The `ShowScrollIndicator` is necessary as I explained before because the `Visible` property is reserved for the scroll bar to handle auto-hide if it is true.

Reply:
> The VisibleChanged event is really not necessary and I will remove it in this PR.

I only need to use the `Parent.VisibleChanged` and `Parent.EnabledChanged` to set the scroll bar visibility and inability if the superview has changed that properties.
Done.

Reply:
> the v2 class should be named ScrollBar

Done.